plat/common/bootinfo_fdt.c

```diff
@@ -198,14 +198,22 @@ void ukplat_bootinfo_fdt_setup(void *fdtp)
 	fdt_bootinfo_fdt_mrd(bi, fdtp);
 	fdt_bootinfo_mem_mrd(bi, fdtp);
 	fdt_bootinfo_initrd_mrd(bi, fdtp);
-	ukplat_memregion_list_coalesce(&bi->mrds);
-
-	/* We use this after coalescing/sorted because this calls
-	 * `ukplat_memregion_alloc()` which would be unsafe to do so before
-	 * knowing that the memory region descriptor list has been coalesced
-	 * and sorted.
-	 */
-	fdt_bootinfo_cmdl_mrd(bi, fdtp);
 
 	bi->dtb = (__u64)fdtp;
 }
+
+/* We use this after coalescing/sorted because this calls
+ * `ukplat_memregion_alloc()` which would be unsafe to do so before
+ * knowing that the memory region descriptor list has been coalesced
+ * and sorted.
+ */
+void ukplat_bootinfo_fdt_post_setup(void)
+{
+	struct ukplat_bootinfo *bi;
+
+	bi = ukplat_bootinfo_get();
+	if (unlikely(!bi))
```

Comment:
Ditto

Reply:
I follow the pattern used in the rest of this file by the original author:
```
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 187) void ukplat_bootinfo_fdt_setup(void *fdtp)
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 188) {
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 189)      struct ukplat_bootinfo *bi;
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 190) 
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 191)      bi = ukplat_bootinfo_get();
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 192)      if (unlikely(!bi))
1afb423425 (Sergiu Moga     2023-05-14 18:21:55 +0300 193)              ukplat_bootinfo_crash("Invalid bootinfo");
```

Reply:
The **unbelievably handsome and charismatic** author may have just changed his mind :thinking:, I am joking (about the latter). In all seriousness though, I believe these changes were back when I barely had any familiarity with this code I guess. I am fine with leaving as is and coming back when we rearch ukboot.